c7n/resources/appelb.py

```diff
@@ -153,33 +140,119 @@ def process_alb(self, alb):
 
 
-@filters.register('is-ssl')
-class AppELBIsSSLFilter(AppELBListenerFilter):
+class AppELBAttributeFilter(Filter):
+    """ Base class for filters that query LB attributes.
+    """
+    def initialize(self, albs):
+        def _process_attributes(alb):
+            if 'Attributes' not in alb:
+                client = local_session(
+                    self.manager.session_factory).client('elbv2')
+                results = client.describe_load_balancer_attributes(
+                    LoadBalancerArn=alb['LoadBalancerArn'])
+                alb['Attributes'] = results['Attributes']
+
+        with self.manager.executor_factory(max_workers=2) as w:
+            list(w.map(_process_attributes, albs))
+
+
+class AppELBTargetGroupFilter(Filter):
+    """ Base class for filters that query LB target groups.
+    """
+    def initialize(self, albs):
+        def _process_target_groups(alb):
+            if 'TargetGroups' not in alb:
+                client = local_session(
+                    self.manager.session_factory).client('elbv2')
+                results = client.describe_target_groups(
+                    LoadBalancerArn=alb['LoadBalancerArn'])
+                alb['TargetGroups'] = results['TargetGroups']
+
+                for target_group in alb['TargetGroups']:
+                    result = client.describe_target_health(
+                        TargetGroupArn=target_group['TargetGroupArn'])
+                    target_group['TargetHealthDescriptions'] = result[
+                        'TargetHealthDescriptions']
+
+        with self.manager.executor_factory(max_workers=2) as w:
+            list(w.map(_process_target_groups, albs))
+
+
+@filters.register('is-https')
+class AppELBIsHTTPSFilter(AppELBListenerFilter):
+    """
+    """
 
-    schema = type_schema('is-ssl')
+    schema = type_schema('is-https')
 
     def process(self, albs, event=None):
-        def _is_ssl(alb):
+        def _is_https(alb):
             for listener in alb.get('Listeners', []):
                 if listener['Protocol'] == 'HTTPS':
                     return True
             return False
 
-        self.describe_appelb_listeners(albs)
-        return [alb for alb in albs if _is_ssl(alb)]
+        self.initialize(albs)
+        return [alb for alb in albs if _is_https(alb)]
+
+
+@filters.register('healthcheck-protocol-mismatch')
+class AppELBHealthCheckProtocolMismatchFilter(AppELBTargetGroupFilter):
+    """
+    """
+
+    schema = type_schema('healthcheck-protocol-mismatch')
+
+    def process(self, albs, event=None):
+        def _healthcheck_protocol_mismatch(alb):
+            for target_group in alb['TargetGroups']:
+                if (target_group['Protocol'] !=
+                        target_group['HealthCheckProtocol']):
+                    return True
+
+            return False
+
+        self.initialize(albs)
+        return [alb for alb in albs if _healthcheck_protocol_mismatch(alb)]
+
+
+@filters.register('instance-count')
```

Comment:
it feels like the generic value filter already has this covered in conjunction with the value filter derived target group filter, ie what's the delta of this to

```yaml
name: alb-has-instances
resource: app-elb
filters:
  - type: target-group
    key: "TargetHealthDescriptions[]"
    value_type: size
    value: 0
```